packages/sdk/src/util/logger.ts

```diff
@@ -1,2 +1,22 @@
-export type Logger = (message: string) => void;
+export type SimpleLogger = (message: string) => void;
 export const noopLogger = () => {};
+
+export interface IFullLogger {
+  error: SimpleLogger;
+  warn: SimpleLogger;
+  report: SimpleLogger;
+}
+
+export type Logger = SimpleLogger | IFullLogger;
+
+export function toFullLogger(l: Logger): IFullLogger {
+  if ('warn' in l) {
+    return l;
+  } else {
+    return {
+      error: m => l('ERROR: ' + m),
+      warn: m => l('WARNING: ' + m),
+      report: l,
```

Comment:
I think this should be `log` no report so it conforms to global `Console` (and you can just pass in `window.console` as you logger if you really want since it has `error` and `warn` as well).